bulk-update/document-manager/document-manager.js

```diff
@@ -117,18 +117,20 @@ export async function loadDocument(entry, config, fetchFunction = fetch) {
     const stats = fs.statSync(document.markdownFile);
     if (!hasExpired(stats.mtime, mdCacheMs)) {
       document.markdown = fs.readFileSync(document.markdownFile, 'utf8');
-      reporter.log('load', 'success', 'Loaded markdown', document.markdownFile);
+      reporter.log('load', 'success', 'Loaded markdown', { entry });
     }
   }
 
   if (!document.markdown) {
     document.markdown = await fetchMarkdown(`${document.url}.md`, reporter, fetchWaitMs, fetchFunction);
-    reporter.log('load', 'success', 'Fetched markdown', `${document.url}.md`);
-
-    if (document.markdown && mdDir) {
-      const folder = document.markdownFile.split('/').slice(0, -1).join('/');
-      fs.mkdirSync(folder, { recursive: true });
-      fs.writeFileSync(document.markdownFile, document.markdown);
+    if (document.markdown) {
+      reporter.log('load', 'success', 'Fetched markdown', { entry });
+
+      if (mdDir) {
+        const folder = document.markdownFile.split('/').slice(0, -1).join('/');
+        fs.mkdirSync(folder, { recursive: true });
+        fs.writeFileSync(document.markdownFile, document.markdown);
+      }
     }
   }
```

Comment:
Yeah this can be cleaned up, returning early means we have the markdown and this doesn't represent that.
What could be useful in the future is having a document class to encapsulate some of this logic but for now I'll add a load and a save function to make this easier to read.